hoodie-client/src/main/java/com/uber/hoodie/HoodieWriteClient.java

```diff
@@ -519,7 +520,8 @@ public boolean commit(String commitTime,
     }
 
     try {
-      String actionType = table.getCommitActionType();
+      String actionType = table.getActiveTimeline().getCommitsTimeline().filterInflights()
```

Comment:
@vinothchandar This is a change required to support COMMIT, DELTA_COMMIT and COMPACTION/COMMIT from the table. Please take a look at this.

Reply:
could you expand? its not obvious to me, sorry

Reply:
Yeah, so we figure out commitAction based on the tableType. The commit action for COW -> COMMIT and for MOR -> DELTA_COMMIT. But when we do compaction on MOR tabletype we want action type to be COMMIT. So, table.getCommitActionType() needs to be smarter. For now, I basically look at the current inflight commit and figure out the actiontype based on that (since during startCommit() and startCompaction() we create different action types). 

The other option is to create a commitCompaction() and re-use some code from commt(). Athough we want to go away from the notion of compaction(..) usage for users but i think with async compaction and all, it might be clear to have that terminology.

Reply:
Figuring out action type dynamically sounds a bit scary to me. any simpler alternatives ? is the method/code cleanup you are suggesting a better way?

Reply:
Instead of calling commit(..), clients will have to call commitCompaction(..) and then we share the code for both these with the actionType hard coded to COMMIT in commitCompaction(..), we could think about this approach.

Reply:
lets do that for now. seems safer. with dynamically determining, worried about some corner cases since we are assuming stuff about the timeline.

Reply:
yeah, okay, let me make that change.